subsys/zbus/Kconfig

```diff
@@ -25,13 +25,13 @@ config ZBUS_MSG_SUBSCRIBER
 
 if ZBUS_MSG_SUBSCRIBER
 
-choice
+choice ZBUS_MSG_SUBSCRIBER_BUF_ALLOC
 	prompt "ZBus msg_subscribers buffer allocation"
```

Comment:
Right, added.

Reply:
@rodrigopex I'm wondering if we should set `ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_DYNAMIC` as a default choice since dynamic allocation is not enabled by default in Zephyr ... Maybe we should do something like that:

<pre>
choice ZBUS_MSG_SUBSCRIBER_BUF_ALLOC
	prompt "ZBus msg_subscribers buffer allocation"
	default ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_DYNAMIC if HEAP_MEM_POOL_SIZE > 0
	default ZBUS_MSG_SUBSCRIBER_BUF_ALLOC_STATIC
</pre>

What is your opinion here?

Reply:
The static message subscriber buffer is a "fat" net_buf that must have room enough for all message sizes. It is not memory-optimized. If you have one single message with 256 bytes, the fat buffer must have at least that. It should be used only in cases where dynamic allocation is not permitted, such as MISRA-C and other scenarios. In that case, the best solution for common usage is HEAP memory. Have you seen the message subscriber implementation? What do you think of the approach with static allocation?

Reply:
The message subscriber is not available by default. I do not see a problem here. 

>  It would be good to hear other people's opinions in this area

Go ahead.

The Message Subscriber PR history provides part of the answer. The Nordic Event Manager and ZMK distribute messages similarly, so HEAP is the way to go in this situation. Please review the PR and the code before questioning the decisions. Thank you.